shell/browser/notifications/mac/cocoa_notification.mm

```diff
@@ -60,68 +66,160 @@
 
   [notification_ setHasActionButton:false];
 
-  int i = 0;
-  action_index_ = UINT_MAX;
   NSMutableArray* additionalActions =
       [[[NSMutableArray alloc] init] autorelease];
+
+  NSString* replyKey_ = @"";
   for (const auto& action : options.actions) {
-    if (action.type == u"button") {
-      if (action_index_ == UINT_MAX) {
-        // First button observed is the displayed action
-        [notification_ setHasActionButton:true];
-        [notification_
-            setActionButtonTitle:base::SysUTF16ToNSString(action.text)];
-        action_index_ = i;
-      } else {
-        // All of the rest are appended to the list of additional actions
-        NSString* actionIdentifier =
-            [NSString stringWithFormat:@"%@Action%d", identifier, i];
-        NSUserNotificationAction* notificationAction = [NSUserNotificationAction
-            actionWithIdentifier:actionIdentifier
-                           title:base::SysUTF16ToNSString(action.text)];
-        [additionalActions addObject:notificationAction];
-        additional_action_indices_.insert(
-            std::make_pair(base::SysNSStringToUTF8(actionIdentifier), i));
-      }
-    }
-    i++;
+    // SAP-15908 - Refine notification-activation for cold start in macOS
+    if (action.type == electron::NotificationAction::sTYPE_TEXT)
+      replyKey_ = [[NSString alloc]
+          initWithBytes:action.arg.data()
+                 length:action.arg.length() * sizeof(action.arg[0])
+               encoding:NSUTF16LittleEndianStringEncoding];
+    // SAP-15259 - Add the reply field on a notification
+    if (action.type != electron::NotificationAction::sTYPE_BUTTON)
```

Comment:
The purpose of this have earlier skip for circle for types such as TEXT (need for reply) we want to avoid adding replies here additional_action_indices_ . Do not see problems with earlier circle termination, yet. In my vision current solution is smaller that back logic for == BUTTON